testes automatizados (#2)
betrybe/project-cookmaster: README.md

# Boas vindas ao repositório do projeto Cookmaster!

Você já usa o GitHub diariamente para desenvolver os exercícios, certo? Agora, para desenvolver os projetos, você deverá seguir as instruções a seguir. Fique atento a cada passo, e se tiver qualquer dúvida, nos envie por Slack! #vqv 🚀

Aqui você vai encontrar os detalhes de como estruturar o desenvolvimento do seu projeto a partir deste repositório, utilizando uma branch específica e um Pull Request para colocar seus códigos.

---

## Instruções para entregar seu projeto:

### ANTES DE COMEÇAR A DESENVOLVER:

1. Clone o repositório

>   * `git clone https://github.com:tryber/sd-0x-cookmaster.git`.

Reviewer: Não se esqueça de que o nome do repositório tem que ser alterado pra bater com esse aqui!

Reviewer: @brunobatista25 Falta isso aqui! =)

- Utilize `includes` do EJS para renderizar a navbar das páginas

- Parte do HTML ficará repetido em todas as páginas como, por exemplo, a barra de navegação.

- Para esses conteúdos repetitivos, você pode utilizar `includes` do EJS.

- A [documentação do EJS](https://ejs.co/#docs) (dê um Ctrl + F e pesquise por "includes") fala brevemente sobre o use de includes nas suas views.

### Data de Entrega

O projeto tem até a seguinte data: `DD/MM/YYYY - 14:00h`. Para ser entregue a avaliação final.

---

## Requisitos Obrigatórios:

### 💡Veja o exemplo a seguir de como o projeto pode se parecer depois de pronto. Lembre-se que você pode ~~e deve~~ ir além para deixar o projeto com a sua cara e impressionar à todos!

Você pode acessar um protótipo da aplicação com todas as telas (tanto obrigatórias quanto bônus) [neste link](https://www.figma.com/file/CAEkOBX1n3mpVXr4kjgvY8/Project-Cookmaster?node-id=0%3A1).

Não estamos avaliando o **estilo** da página. Cores, tamanhos de texto e afins não serão avaliados.

## ⚠️ Leia-os atentamente e siga à risca o que for pedido. ⚠️

> Iremos apenas pedir que adicone data-testid em alguns elemetos, e algumas tags e textos especificos, então siga a risca o que iremos pedir nos requisitos, para que a automação funcione.

Reviewer: ```suggestion
Vamos pedir que você adicione um `data-testid` em alguns elementos, além de algumas tags e textos específicos. Siga à risca o que iremos pedir nos requisitos, para que o corretor automatizado possa avaliar o seu projeto corretamente!
```

O não cumprimento de um requisito, total ou parcialmente, impactará em sua avaliação.

---

### 👀Observações importantes:

> Haverá um arquivo chamado `cookmaster.sql` onde já contém a criação de do banco e alguns inserts.

Reviewer: ```suggestion
O arquivo chamado `cookmaster.sql` já contém o código para criação de do banco e insere alguns dados iniciais.
```

Reviewer: Não quis mergear essa correção?

> Para rodar o arquivo basta rodar o comando 

Reviewer: ```suggestion
Para rodar o arquivo basta rodar o comando:
```

`mysql -u root -p`

> Isso fará com que abra o terminal do mysql depois basta executar o comando:

Reviewer: ```suggestion
Isso fará com que o terminal do MySQL se abra. Depois, basta executar o comando:
```

`source pasta_do_projeto/cookmaster.sql`

> Assim já irá criar o banco e alguns inserts.

Reviewer: ```suggestion
Assim já irá criar o banco e terá alguns dados inseridos. **É essencial seguir esses passos!**
```

#### Funcionalidades de visualização

> As páginas abaixo podem ser acessadas sem login:

* Listagem de receitas
* Tela para visualizar uma receita específica
* Página de cadastro de usuários

### 1 - Crie uma tela de listagem de receitas.

A página deve ser acessível através da rota principal (`/`).

Para cada receita, deve ser mostrado apenas o nome da receita e o nome da pessoa que cadastrou aquela receita, bem como um link para ver seus detalhes.

Um botão "Nova receita" deve ser exibido **apenas quando houver um usuário logado**.

Tela home:

> O página deverá ter o texto 'Cookmaster'.

Reviewer: Ó a concordância e a formatação! Importante o texto estar bem bonitinho :)

```suggestion
- A página deverá ter o texto 'Cookmaster'.
```

Dei essa suggestion aqui mas há vários lugares para fazer ajustes similares. Rola?

O página deverá ter o texto 'Receitas'.

> O link 'Ver mais' deverá ser utilizado o elemento `<a>Ver mais</a>`

Reviewer: ```suggestion
O link 'Ver mais' deverá estar num elemento `<a>` com o texto `Ver mais`. Algo na linha de: `<a href="Link">Ver mais</a>`
```

![Tela Home](./public/login.png)

#### As seguintes verificações serão feitas:

**[Verificar se estou na home e tem os títulos 'Cookmaster' e 'Receitas'.]**

**[Verificar se não existe o botão 'Nova Receita' quando acesso home sem estar logado.]**

**[Verificar se existe o botão 'Nova Receita' quando estou logado e acessei a home.]**

**[Verificar se existe receita na tela com 'nome da receita', 'nome do usuário' e o link da receita 'Ver mais'.]**

### 2 - Crie uma página de cadastro de usuários.

Um usuário precisa ter os campos ID, E-mail, Senha, Nome e Sobrenome. Todos os campos são obrigatórios. O ID deve ser gerado automaticamente, não devendo ser preenchido pelo usuário no momento do cadastro.

A validação dos campos deve acontecer no backend, e uma mensagem deve ser enviada ao frontend através de uma propriedade passada para o EJS, da mesma forma que acontece com a view `users/login`.

**⚠️ Atenção ⚠️**: O sistema de autenticação espera que as funções `findUserByEmail` e `findUserById` retornem um objeto com, pelo menos, os campos `email`, `password` e `id`. Se você alterar o nome desses campos, precisará alterar o código de login.

> Adicione na tela de login o botão ou link 'Cadastre-se' e deverá conter a tag data-testid="cadastrar".

Reviewer: ```suggestion
Adicione na tela de login o botão ou link 'Cadastre-se' que deverá conter a tag `data-testid="cadastrar"`.
```

- O middleware `authMiddleware` recebe um único parâmetro chamado `required`. Quando verdadeiro, esse parâmetro faz com que uma rota só possa ser acessada por pessoas logadas. Em alguns casos, uma página precisa ter acesso aos dados do usuário, caso essa pessoa esteja logada, mas deve continuar podendo ser acessada por pessoas que não estejam autenticadas. Para esse caso, passe `false` no parâmetro required para a função `authMiddleware`. Ex.: `middlewares.auth(false)`.

- Utilize `includes` do EJS para renderizar a navbar das páginas

- Parte do HTML ficará repetido em todas as páginas como, por exemplo, a barra de navegação.

- Para esses conteúdos repetitivos, você pode utilizar `includes` do EJS.

> - A [documentação do EJS](https://ejs.co/#docs) (dê um Ctrl + F e pesquise por "includes") fala brevemente sobre o use de includes nas suas views.

Reviewer: ```suggestion
- A [documentação do EJS](https://ejs.co/#docs) (dê um `Ctrl + F` e pesquise por "includes") fala brevemente sobre o use de includes nas suas views.
```

### Data de Entrega

O projeto tem até a seguinte data: `DD/MM/YYYY - 14:00h`. Para ser entregue a avaliação final.

---

## Requisitos Obrigatórios:

### 💡Veja o exemplo a seguir de como o projeto pode se parecer depois de pronto. Lembre-se que você pode ~~e deve~~ ir além para deixar o projeto com a sua cara e impressionar à todos!

Você pode acessar um protótipo da aplicação com todas as telas (tanto obrigatórias quanto bônus) [neste link](https://www.figma.com/file/CAEkOBX1n3mpVXr4kjgvY8/Project-Cookmaster?node-id=0%3A1).

Não estamos avaliando o **estilo** da página. Cores, tamanhos de texto e afins não serão avaliados.

## ⚠️ Leia-os atentamente e siga à risca o que for pedido. ⚠️

> Vamos pedir que você adicone `data-testid` em alguns elemetos,além de algumas tags e textos especificos, Siga à risca o que iremos pedir nos requisitos, para que o corretor automatizado possa avaliar o seu projeto corretamente.

Reviewer: ```suggestion
Vamos pedir que você adicione `data-testid` em alguns elementos, além de algumas tags e textos especificos. Siga à risca o que iremos pedir nos requisitos, para que o corretor automatizado possa avaliar o seu projeto corretamente.
```

O não cumprimento de um requisito, total ou parcialmente, impactará em sua avaliação.

---

### 👀Observações importantes:

Haverá um arquivo chamado `cookmaster.sql` onde já contém a criação de do banco e alguns inserts.

Para rodar o arquivo basta rodar o comando:

`mysql -u root -p`

Isso fará com que abra o terminal do MySQL se abra. Depois, basta executar o comando:

`source pasta_do_projeto/cookmaster.sql`

Assim já irá criar o banco e terá alguns dados inseridos. **É essencial seguir esses passos!**

> Haverá um arquivo no seguinte caminho `sd-0x-cookmaster/cypress/plugins/index.js`, Neste arquivo na linha 17.
> 
> Haverá a seguinte comando:

Reviewer: ```suggestion
Haverá um arquivo no caminho: `sd-0x-cookmaster/cypress/plugins/index.js`. Neste arquivo, na linha 17, haverá a seguinte comando:
```

> `const connection = my.createConnection({host: process.env.HOSTNAME, user: process.env.MYSQL_USER, password: process.env.MYSQL_PASSWORD});`

Reviewer: ```suggestion
`const connection = my.createConnection({ host: process.env.HOSTNAME, user: process.env.MYSQL_USER, password: process.env.MYSQL_PASSWORD });`
```

> Você irá precisar configurar as variáveis globais do myqsl, você pode usar esse [Variáveis com nodeJs](https://blog.rocketseat.com.br/variaveis-ambiente-nodejs/) como exemplo, outra opcão, é você mudar manualmente como esse exemplo:

Reviewer: ```suggestion
**Você irá precisar configurar as variáveis globais do MySQL.** Você pode usar esse [conteúdo de variáveis de ambiente com NodeJS](https://blog.rocketseat.com.br/variaveis-ambiente-nodejs/) como referência. Outra opcão é você mudá-las manualmente, como neste exemplo:
```

> `const connection = my.createConnection({host: '127.0.0.1, user: 'root', password: '12345678'});`

Reviewer: ```suggestion
`const connection = my.createConnection({ host: '127.0.0.1, user: 'root', password: '12345678' });`
```

Fica da sua escolha como inserir esses valores.

#### Funcionalidades de visualização

> As páginas abaixo podem ser acessadas sem login:

* Listagem de receitas
* Tela para visualizar uma receita específica

> * Página de cadastro de usuários

Reviewer: ```suggestion
* Página de cadastro de pessoas usuárias
```

### 1 - Crie uma tela de listagem de receitas.

A página deve ser acessível através da rota principal (`/`).

Para cada receita, deve ser mostrado apenas o nome da receita e o nome da pessoa que cadastrou aquela receita, bem como um link para ver seus detalhes.

Um botão "Nova receita" deve ser exibido **apenas quando houver um usuário logado**.

### Tela da home:

- A página deverá ter o texto 'Cookmaster'.

- A página deverá ter o texto 'Receitas'.

- O link 'Ver mais' deverá estar num elemento `<a>` com o texto `Ver mais`. Algo na linha de: `<a href="Link">Ver mais</a>`

- O botão 'Nova Receita' deverá conter o `data-testid="nova-receita"`

> (Esse botão deverá ficar visível na tela apenas estando logado.)

Reviewer: ```suggestion
(Esse botão deverá ficar visível na tela apenas estando logado).
```

- O botão 'Buscar Receita' deverá conter o `data-testid="buscar-receita"`
(Esse botão deverá ficar visível na tela estando logado ou não)

- O 'Login' deverá conter a tag `data-testid="login"`

- O 'Logout' deverá conter a tag `data-testid="logout"`

![Tela Home](./public/home_nao_autenticado.png)

### Tela de login:

- A página deve ser acessível através da rota principal (`/login`). E deverá ser acessada pelo 'Login' da página Home.

- O campo 'Email' deverá conter a tag `data-testid="email"`

- O campo 'Senha' deverá conter a tag `data-testid="senha"`

- O botão 'Login' deverá conter a tag `data-testid="entrar"`

![Tela Home](./public/login.png)

#### As seguintes verificações serão feitas:

**[Verificar se estou na home e tem os títulos 'Cookmaster' e 'Receitas']**

**[Verificar se não existe o botão 'Nova Receita' quando acesso home sem estar logado]**

**[Verificar se existe o botão 'Nova Receita' quando estou logado e acessei a home]**

> **[Verificar se existe receita na tela com 'nome da receita', 'nome do usuário' e o link da receita 'Ver mais']**

Reviewer: ```suggestion
**[Verificar se existe, na tela, uma receita previamente cadastrada com 'nome da receita', 'nome do usuário' e o link da receita em 'Ver mais']**
```

### 2 - Crie uma página de cadastro de usuários.

> Um usuário precisa ter os campos ID, E-mail, Senha, Nome e Sobrenome. Todos os campos são obrigatórios. O ID deve ser gerado automaticamente, não devendo ser preenchido pelo usuário no momento do cadastro.

Reviewer: ```suggestion
Uma pessoa usuária precisa ter preenchidos os campos `ID`, `E-mail`, `Senha `, `Nome` e `Sobrenome`. Todos os campos são obrigatórios. O `ID` deve ser gerado automaticamente, não devendo ser preenchido pela pessoa usuária no momento do cadastro.
```

> A validação dos campos deve acontecer no backend, e uma mensagem deve ser enviada ao frontend através de uma propriedade passada para o EJS, da mesma forma que acontece com a view `users/login`.

Reviewer: ```suggestion
A validação dos campos deve acontecer no back-end, e uma mensagem deve ser enviada ao front-end através de uma propriedade passada para o EJS, da mesma forma que acontece com a view `users/login`.
```

**⚠️ Atenção ⚠️**: O sistema de autenticação espera que as funções `findUserByEmail` e `findUserById` retornem um objeto com, pelo menos, os campos `email`, `password` e `id`. Se você alterar o nome desses campos, precisará alterar o código de login.

Adicione na tela de login o botão ou link 'Cadastre-se' que deverá conter a tag `data-testid="cadastrar"`

### Tela de Cadastro:

- O campo 'Email' deverá conter a tag `data-testid="email"`

- O campo 'Senha' deverá conter a tag `data-testid="senha"`

- O campo 'Confirmar Senha' deverá conter a tag `data-testid="confirmar-senha"`

- O campo 'Nome' deverá conter a tag `data-testid="nome"`

- O campo 'Sobrenome' deverá conter a tag `data-testid="sobrenome"`

- O botão 'Cadastar-se' deverá conter a tag `data-testid="cadastrar"`

- O botão 'Login' deverá conter a tag `data-testid="login"`

![Tela Cadastro de usuário](./public/cadastro.png)

#### As seguintes verificações serão feitas:

> **[Criar um usuário com sucesso]**

Reviewer: ```suggestion
**[Cadastro de uma pessoa usuária com sucesso]**
```

> - Ao cadastar o usário deverá aparecer a mensagem "Cadastro efetuado com sucesso!"

Reviewer: ```suggestion
- Ao se cadastrar uma pessoa usuária, deverá aparecer a mensagem "Cadastro efetuado com sucesso!"
```

> **[Validar o campo email]**

Reviewer: ```suggestion
**[Validação o campo "Email"]**
```

> - Ao cadastar um email inválido deverá aparecer a mensagem "O email deve ter o formato email@mail.com"

Reviewer: ```suggestion
- Ao se cadastrar um email inválido, deverá aparecer a mensagem "O email deve ter o formato email@mail.com"
```

> **[Validar o campo senha]**

Reviewer: ```suggestion
**[Validação do campo "Senha"]**
```

- Ao cadastar uma senha inválida deverá aparecer a mensagem "A senha deve ter pelo menos 6 caracteres"

> **[Validar campos confirmar senha]**

Reviewer: ```suggestion
**[Validação do campo "Confirmar Senha"]**
```

> - Ao cadastar um confirmar senha inválida deverá aparecer a mensagem "As senhas tem que ser iguais"

Reviewer: ```suggestion
- Ao se inserir um "confirmar senha" inválido, deverá aparecer a mensagem "As senhas tem que ser iguais"
```

> **[Validar campos nome]**

Reviewer: ```suggestion
**[Validação do campo "Nome"]**
```

- Ao cadastar um nome inválido deverá aparecer a mensagem "O primeiro nome deve ter, no mínimo, 3 caracteres, sendo eles apenas letras"

> **[Validar campos sobrenome]**

Reviewer: ```suggestion
**[Validação do campo "Sobrenome"]**
```

> - Ao cadastar um sobrenome inválido deverá aparecer a mensagem "O segundo nome deve ter, no mínimo, 3 caracteres, sendo eles apenas letras"

Reviewer: ```suggestion
- Ao se cadastrar um sobrenome inválido, deverá aparecer a mensagem "O segundo nome deve ter, no mínimo, 3 caracteres, sendo eles apenas letras"
```

> Dica: esse é um dos casos no qual você pode utilizar o `authMiddleware` passando `false` para o parâmetro `required`, e passar o conteúdo de `req.user` para a view, o que o permitirá determinar se existe um usuário logado e, portanto, se os botões devem ser exibidos.

### Tela sem estar logado:

![Tela Receita especifica](./public/receita_especifica.png)

> ### Tela com usuário logado:

Reviewer: ```suggestion
### Tela com a pessoa usuária logada:
```

- O botão 'Editar Receita' deverá conter a tag `data-testid="editar-receita"`

- O botão 'Excluir Receita' deverá conter a tag `data-testid="excluir-receita"`

> ![Tela Receita especifica](./public/receita_especifica_autenticada.png)

Reviewer: ```suggestion
![Tela de Receita Especifica](./public/receita_especifica_autenticada.png)
```

#### As seguintes verificações serão feitas:

> **[Verificar se consigo acessar uma receita específica]**

Reviewer: ```suggestion
**[Verificar se consigo acessar uma receita especifica]**
```

> **[Verificar se consigo acessar uma receita específica depois que faço login]**

Reviewer: ```suggestion
**[Verificar se consigo acessar uma receita especifica depois que faço login]**
```

> **[Quando estou logado verfico se na página de uma receita específica existe os botões 'Editar Receita' e 'Excluir Receita']**

Reviewer: ```suggestion
**[Quando a pessoa está logada, verificar se na página de uma receita específica existem os botões 'Editar Receita' e 'Excluir Receita']**
```

**[Verificar se na página da receita irá exibir o 'título da receita', os 'ingredientes', e a 'forma de preparo' da receita]**

**[Verificar se o botão 'Editar Receita' leva para página de edição de receita]**

**[Verificar se o botão 'Excluir Receita' leva para página de exclusão de receita]**

**[Verificar se quando não estou logado os botões 'Editar Receita' e 'Excluir Receita' não podem estar visíveis na tela]**

### 4 - Cria uma página de buscar de receitas.

A página deve estar acessível através do endpoint `/recipes/search`.

Um input do tipo texto deve ser exibido juntamente com um botão "Buscar". O conteúdo do input deve ser enviado para o endpoint `GET /recipes/search` através do parâmetro `q` na query string.

> No backend, o valor do input de texto estará acessível através da propriedade `q` do objeto `req.query`. Caso nada seja informado para pesquisa, a view deve ser renderizada apenas com o campo de pesquisa. Caso um valor seja informado, uma lista semelhante à tela de listar receitas deve ser exibida, contendo o título, nome da pessoa que cadastrou, e um link para cada receita.

Reviewer: ```suggestion
No back-end, o valor do input de texto deverá estar acessível através da propriedade `q` do objeto `req.query`. Caso nada seja informado para pesquisa, a view deverá ser renderizada apenas com o campo de pesquisa. Caso um valor seja informado, uma lista semelhante à tela de listar receitas deve ser exibida, contendo o título, nome da pessoa que cadastrou, e um link para cada receita.
```

Para realizar a pesquisa, o controller de receitas deve solicitar ao model que pesquise por receitas **contendo em seu nome** o valor digitado no input de pesquisa.

> Na Tela inicial('/') acidione o botão 'Buscar Receitas' deverá conter a tag `data-testid="buscar-receita"`, e ao clicar deverá redirecionar para a página de 'Buscar Receitas'.

Reviewer: ```suggestion
Na Tela inicial('/'), adicione o botão 'Buscar Receitas', que deverá conter a tag `data-testid="buscar-receita"` e, ao clicar-se nela, deverá se redirecionar para a página de 'Buscar Receitas'.
```

> O botão 'Buscar Receitas' ficará sempre visível an tela home estando logado ou não.

Reviewer: ```suggestion
O botão 'Buscar Receitas' ficará sempre visível na tela home, estando a pessoa usuária logada ou não.
```

### Tela de Buscar receitas:

- O campo input deverá conter a tag `data-testid="receita"`

- O botão 'Buscar' deverá conter a tag `data-testid="buscar"`

![Tela Pesquisa Receita](./public/busca_receita.png)

#### As seguintes verificações serão feitas:

**[Verificar se o botão 'Buscar Receitas' redireciona para a página das minhas receitas]**

**[Validar se consigo fazer uma busca de receita]**

**[Validar se não possível buscar uma receita que não existe]**

#### Funções administrativas

> As páginas abaixo só podem ser acessadas fazendo login:

* Cadastro de Receitas
* Edição de Receitas
* Exclusão de Receitas
* Minhas Receitas

> Páginas que **não** podem ser acessadas sem login. Para essas páginas, utilize o `authMiddleware` sem passar parâmetro algum.

### 5 - Crie uma página de cadastro de receitas.

A página deve ser acessível através do endpoint `/recipes/new`, e o formulário deve ser enviado para o endpoint `POST /recipes`

A receita deve ter os campos ID, Nome, Ingredientes, Modo de preparo e Autor. Sinta-se à vontade para modelar o banco da forma que achar melhor. O ID deve ser gerado automaticamente, não devendo ser preenchido no formulário de cadastro de receita.

> O campo dos ingredientes terá que ser um campo input onde pode adicionar varios ingredientes e tambem poderá exlcuir ingredientes. Sendo apresentado em uma lista com um botão 'Excluir Ingrediente' para cada ingrediente.

Reviewer: ```suggestion
O campo dos ingredientes terá que ser um campo input onde se pode adicionar vários ingredientes e também se poderá excluir ingredientes, sendo apresentada uma lista com um botão 'Excluir Ingrediente' para cada ingrediente.
```

**⚠️ Atenção ⚠️**: Lembre-se que a tela não é a única forma de acessar os endpoints. Uma requisição feita utilizando o Postman para o endpoint `POST /recipes/:id` **não deve** alterar o ID da receita ou o nome de quem a cadastrou. Para isso, garanta que não está enviando esses campos ao banco de dados na função de update do seu model de receitas.

Tela de Editar de Receita:

> O campo input 'Nome da Receita' deverá conter a tag data-testid="nome-receita"

Reviewer: ```suggestion
O campo input 'Nome da Receita' deverá conter a tag `data-testid="nome-receita"`
```

Em todos os locais onde vc coloca partes de código assim é legal usar as crases pro README ficar mais legível! Que que vc acha?

O campo input 'Ingredientes' deverá conter a tag data-testid="ingredientes".

O botao 'Adicionar Ingrediente' deverá conter a tag data-testid="adicionar-ingrediente"

O campo input 'Modo de Preparo' deverá conter a tag data-testid="modo-de-preparo"

O botao 'Salvar Receita' deverá conter a tag data-testid="postar-receita"

![Tela Pesquisa Receita](./public/editar_receita.png)

#### As seguintes verificações serão feitas:

> **[Editar o nome da receita.]**

Reviewer: Rolava de tirar esses pontos finais de dentro dos colchetes, será que não? 😬

**[Editar os ingredientes da receita.]**

**[Editar o modo de preparo da receita.]**

### 7 - Crie uma página de exclusão de uma receita.

A página deve ser acessível através do endpoint `/recipes/:id/delete`, e só pode ser acessada pela pessoa que cadastrou a receita.

Ao acessar a página, um formulário deve ser exibido, solicitando a senha da pessoa para confirmar a operação. Esse formulário deve ser enviado para o endpoint `POST /recipes/:id/delete`.

A receita só deve ser excluída caso a senha esteja correta. Caso ela esteja incorreta, a pessoa deve ser redirecionada à página de exclusão da receita com a mensagem "Senha incorreta.".

Caso a receita seja excluída com sucesso, a pessoa deve ser redirecionada à página de listagem de receitas.

> Tela de Excluir Receita:

Reviewer: Rola de dar um destaque pras seções de cada tela. Que que vc acha?

#### As seguintes verificações serão feitas:

**[Editar o nome da receita]**

**[Editar os ingredientes da receita]**

**[Editar o modo de preparo da receita]**

### 7 - Crie uma página de exclusão de uma receita.

A página deve ser acessível através do endpoint `/recipes/:id/delete`, e só pode ser acessada pela pessoa que cadastrou a receita.

Ao acessar a página, um formulário deve ser exibido, solicitando a senha da pessoa para confirmar a operação. Esse formulário deve ser enviado para o endpoint `POST /recipes/:id/delete`.

A receita só deve ser excluída caso a senha esteja correta. Caso ela esteja incorreta, a pessoa deve ser redirecionada à página de exclusão da receita com a mensagem "Senha incorreta.".

Caso a receita seja excluída com sucesso, a pessoa deve ser redirecionada à página de listagem de receitas.

### Tela de Excluir Receita:

- O campo input 'Senha' deverá conter a tag `data-testid="senha"`

- O botão 'Confirmar' deverá conter a tag `data-testid="confirmar"`

- O botão 'Cancelar' deverá conter a tag `data-testid="cancelar"`

![Tela Excluir Receita](./public/excluir_receita.png)

#### As seguintes verificações serão feitas:

> **[Tentar excluir uma receita passando a senha errada e verificar a mensagem de erro]**

Reviewer: ```suggestion
**[Tentar excluir uma receita passando a senha errada e validar a mensagem de erro]**
```

Ao inserir a senha errada deverá mostrar a mensagem 'Senha Incorreta.'

> **[Excluir receita com sucesso e verificar se foi redirecionada à página de listagem de receitas]**

Reviewer: ```suggestion
**[Excluir receita com sucesso e verificar se foi houve redirecionamento à página de listagem de receitas]**
```

> Lembrete: o ID do usuário logado está disponível em `req.user.id`.

Na Tela inicial('/') acidione o botão 'Minhas Receitas' deverá conter a tag `data-testid="minhas-receitas"`, e ao clicar deverá redirecionar para a página de 'Minhas Receitas'. E só poderá ser visualizado caso esteja logado.

### Tela de Minhas Receitas:

![Tela minhas Receita](./public/minhas_receitas.png)

#### As seguintes verificações serão feitas:

> **[Validar se o botão minhas receitas está redirecionando para página das minhas receitas]**

Reviewer: ```suggestion
**[Validar se o botão 'Minhas Receitas' está redirecionando para página das minhas receitas]**
```

O ID da pessoa não deve poder ser editado. Nem através da tela, nem através de uma request realizada pelo Postman. Para isso, garanta que seu model não envia esse campo para o banco de dados.

Na Tela inicial('/') acidione o botão 'Minha Conta' deverá conter a tag `data-testid="minha-conta"`, e ao clicar deverá redirecionar para a página de 'Editar Usuário'. E só poderá ser visualizado caso esteja logado.

### Tela de Editar Usuario:

- O campo 'Email' deverá conter a tag `data-testid="email"`

- O campo 'Senha' deverá conter a tag `data-testid="senha"`

- O campo 'Confirmar Senha' deverá conter a tag `data-testid="confirmar-senha"`

- O campo 'Nome' deverá conter a tag `data-testid="nome"`

> - O campo 'Sobrenome' deverá conter a tag `data-testid="sobrenome"`

Reviewer: É bom especificar o elemento que deverá ter a tag, será que não? Imagino vc precisando que o id fique no `input` e alguém coloca na `label` do campo, bota fé?

Author: opa esse ai faltou colocar input

> - O botão 'Cadastar-se' deverá conter a tag `data-testid="cadastrar"`

Author: opa, ajustado pra salvar
